### PR TITLE
Reverting Changes - closes #11

### LINKED ISSUE #11: Reverting an unsaved reminder
When actively editing a reminder: If the reminder has unsaved changes, the user will see a "Revert" button. When clicked it will automatically rollback the attributes of the reminder to its original, clean state.

**Pro tip**: I you're doing something that seems hard, maybe you should check the Ember Data API docs?

### PR DESCRIPTION
**Pull request by Jeff-Duke**
## Purpose

This feature allows the user to click an undo button and revert any changes they've made to a reminder. 

## Approach

When a user edits a reminder they will now see an 'Undo' button.  The button is disabled by default.  If they edit any of the fields the button will enable.  Clicking on the button reverts all changes back to the original values. 

### Learning

Looked through the ember docs and found guides for finding a record, checking to see if it's dirty and rolling back any dirty props.  

### Test coverage 
Created an acceptance test that opens the first reminder, stores the reminder title in a variable, verifies the undo button is disabled, changes the title, verifies the button is enabled, clicks the revert-changes button and verifies the current reminder title is equal to the original title that was stored in the variable. 

@stevekinney @brittanystoroz @martensonbj

### CODE REVIEW

File: app/controllers/reminders/reminder.js [hunk 1]
```diff
@@ -9,12 +9,26 @@ export default Ember.Controller.extend({
     saveReminder(reminderForm) {
       let reminder = reminderForm.getProperties('title', 'date', 'notes', 'id');
       reminder.date = new Date(reminder.date);
-      this.get('store').findRecord('reminder', reminder.id).then(function(targetReminder){
-        targetReminder.setProperties({title: reminder.title, date: reminder.date, notes: reminder.notes});
+      this.get('store').findRecord('reminder', reminder.id).then(function(targetReminder) {
```
Comment (stevekinney) [inline app/controllers/reminders/reminder.js]: What is the purpose of this code? It seems way more complicated than it needs to be. We definitely should _not_ need to fire off a whole new AJAX call to get model data from the server in a controller. That is the role of of the model hook in the route.

File: app/controllers/reminders/reminder.js [hunk 2]
```diff
@@ -9,12 +9,26 @@ export default Ember.Controller.extend({
...
+        targetReminder.setProperties({
+          title: reminder.title,
+          date: reminder.date,
+          notes: reminder.notes
+        });
         targetReminder.save();
       });
       this.toggleProperty('isEditing');
```
Comment (stevekinney) [inline app/controllers/reminders/reminder.js]: This is outside of the promise chain, which means you're going to turn off editing even if the transaction totally blows up.

File: app/controllers/reminders/reminder.js [hunk 3]
```diff
@@ -9,12 +9,26 @@ export default Ember.Controller.extend({
...
+    },
+
+    revertChanges(reminderForm) {
+      console.log('yo button still working', reminderForm.get('hasDirtyAttributes'));
+      let reminder = reminderForm.getProperties('title', 'date', 'notes', 'id');
+      this.get('store').findRecord('reminder', reminder.id).then(targetReminder => {
```
Comment (stevekinney) [inline app/controllers/reminders/reminder.js]: We definitely don't need to fire off an AJAX call for this either.

File: app/templates/reminders/reminder.hbs [hunk 4]
```diff
@@ -3,6 +3,7 @@
 
   {{#if isEditing}}
     {{reminder-form reminder=model formSubmitted=(action 'saveReminder' model) button-text="Save"}}
+    <button class="revert-changes-button"  disabled={{unless model.hasDirtyAttributes 'true'}} {{action 'revertChanges' model}}>Undo</button>
```
Comment (stevekinney) [inline app/templates/reminders/reminder.hbs]: Let's go with one space in between the class and the disabled property. Also, this should _totally_ be inside of the `reminder-form` component.

File: app/controllers/reminders/reminder.js (hunk 1 above)
Comment (Jeff-Duke) [inline app/controllers/reminders/reminder.js]: This code is for updating the record based on what we found in the docs.  https://guides.emberjs.com/v2.8.0/models/creating-updating-and-deleting-records/

The function is pulling in the updated form that the user has edited, pulling off the props and updating them on the record that is on the server.  Is there a better way to do this?

**Pull request merged**